pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/BridgeOutputBuilder.java

```diff
@@ -216,10 +245,53 @@ void fillText(List<OneField> recFields) throws BadRecordException {
         int type = fld.type;
         Object val = fld.val;
         if (DataType.get(type) == DataType.BYTEA) {// from LineBreakAccessor
-            output = new BufferWritable((byte[]) val);
+            if (samplingEnabled) {
+                convertTextDataToLines((byte[]) val);
+            } else {
+                output = new BufferWritable((byte[]) val);
+                outputList.add(output); // TODO break output into lines
+            }
         } else { // from QuotedLineBreakAccessor
             String textRec = (String) val;
             output = new Text(textRec + "\n");
+            outputList.add(output);
+        }
+    }
+
+    void convertTextDataToLines(byte[] val) {
```

Comment:
Add a comment to this function explaining the purpose of this either here or when it is invoked for sampling